### PR TITLE
L42A replacement with M4RA, less balance edition

### PR DESCRIPTION
**Pull request by CapCamIII**
# IMPORTANT NOTES PLEASE READ
THIS DOES NOT CONTAIN THE NEW AMMO AND REBALANCED L42/M4RA THAT #1485 HAD
THIS DOES CONTAIN SOME BALANCE THINGS, BUT IT IS NOT ANYWHERE CLOSE TO THE ABOVE

Also lore team wanted this, plz no gbp close maintainers 🙏 🙏 🧎‍♂️ 🕋 

# About the pull request

Replaces L42A in all marine available sources with the M4RA, the canonical DMR of the USCM, you may notice this is currently the scout rifle, well the scout rifle is now the M4RA Custom, a version that can chamber the HV rounds that are spec ammo, but it can also chamber standard m4ra rounds, albeit doing less damage with them than a normal M4RA.

M4RA has current L42A stats fully, with the three exceptions being having no stock to attach or remove(stock was not integrated it sucked), being able to fit a/vgrip like scout m4RA, which may seem like a huge thing but L42 stats were already insane, so it doesn't effect much.

M4RA Custom(scout gun), gets L42 stats as well, with the exception of having less of a damage mult, meaning when not using the spec ammo, it is out-preformed by the standard m4RA.

Adds new M4RA sprites, both standard and custom, by triiodine 

Adds sprites for all M4RA mag variants, by myself.

This was requested by lore team, previous PR of this with way more balance stuff was #1485
Ok thats about all 🙂 

# Explain why it's good for the game

Lore accuracy is good, and this mostly doesn't effect the actual game outside of the scout rifle changes.
Also scout rifle underpreformed if you weren't omega hell sliming with inc-impact stunlocking while on fire, still will be omega hell slime central but that isn't the thing being solved in this pr , it'll do fine when NOT sliming at least now.

# Testing Photographs and Procedure
It works.

# Changelog

 
:cl:
add: Adds the M4RA as the standard marine DMR, identical stats to L42 with the exception of fitting a v or agrip and no removable stock, stats still the same as l42 without stock.
del: L42 from all marine accessible sources with the exception of black market
balance: Scout M4RA is now the M4RA Custom, it can use standard M4RA magazines but standard M4RA cannot use custom magazines.
balance: Scout M4RA now has L42/M4RA standard stats with the exception of lower damage.
balance: Scout M4RA now can fit magnetic harness, laser sight, however it can no longer fit recoil compensator
fix: R4T sling now has the correct color scheme on LV522
spellcheck: New desc for M4RA and L42 by misty
imageadd: New M4RA icons by triio, both scout and normal
/:cl:

### CODE REVIEW

File: code/datums/supply_packs/ammo.dm [hunk 1]
```diff
@@ -78,6 +78,39 @@
...
+	name = "Magazine box (M4RA, 16x AP mags)"
+	contains = list(
+		/obj/item/ammo_box/magazine/m4ra/ap,
+	)
+	cost = 40
+	containertype = /obj/structure/closet/crate/ammo
+	containername = "\improper M4RA AP magazines crate"
+	group = "Ammo"
+
+/* //commented out because l42 exts were already inaccessable
```
Comment (Zonespace27) [inline code/datums/supply_packs/ammo.dm:103]: commented code is bad, just remove it

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/rifles.dm [hunk 2]
```diff
@@ -1324,13 +1324,73 @@
...
+	accuracy_mult = BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_5
+	accuracy_mult_unwielded = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_4
+	damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_6
+	recoil_unwielded = RECOIL_AMOUNT_TIER_4
+	damage_falloff_mult = 0
+	scatter = SCATTER_AMOUNT_TIER_8
+
+/obj/item/weapon/gun/rifle/m4ra/handle_starting_attachment()
+	..()
+	var/obj/item/attachable/m4ra_barrel/S = new(src)
```
[the remark this answers is not in the document]
Comment (CapCamIII) [reply code/modules/projectiles/guns/rifles.dm:1379]: this is copy-pasted from every single integrated attachment in the game
🗼

**Comment by MusselMain** [comment]
Gonna need to change L42A, and ABR-40 stats and description

**Comment by CapCamIII** [comment]
this is literally completely up to date what do you mean merge conflict github-actions [BOT]

**Comment by NewyearnewmeUwu** [comment]
It looks a little comically large ngl.

File: code/modules/projectiles/gun_attachables.dm [hunk 3]
```diff
@@ -1511,6 +1511,59 @@ Defined in conflicts.dm of the #defines folder.
 	pixel_shift_y = 16
 	hud_offset_mod = 10 //A sprite long enough to touch the Moon.
 
+/obj/item/attachable/m4ra_barrel
```
Comment (Segrain) [inline code/modules/projectiles/gun_attachables.dm:1514]: Do not do this. If the only purpose of its existence is to modify the sprite, simply _change the sprite_.

**Review by Segrain — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/specialist.dm [hunk 4]
```diff
@@ -516,55 +516,82 @@
...
-	fire_delay = FIRE_DELAY_TIER_6
+	fire_delay = FIRE_DELAY_TIER_8
 	burst_amount = BURST_AMOUNT_TIER_2
 	burst_delay = FIRE_DELAY_TIER_10
-	accuracy_mult = BASE_ACCURACY_MULT
+	accuracy_mult = BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_5
 	scatter = SCATTER_AMOUNT_TIER_8
 	burst_scatter_mult = SCATTER_AMOUNT_TIER_8
-	damage_mult = BASE_BULLET_DAMAGE_MULT
+	damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_4
```
Comment (Segrain) [inline code/modules/projectiles/guns/specialist.dm:590]: By doing this you completely overtuned how much DPS scout gun has with its special ammunition. That is not the meaning of "less balance" I expected, though it is technically correct.

**Review by Segrain — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun_attachables.dm (hunk 3 above)
Comment (CapCamIII) [reply code/modules/projectiles/gun_attachables.dm:1514]: Handled on discord

File: code/modules/projectiles/guns/specialist.dm (hunk 4 above)
Comment (CapCamIII) [reply code/modules/projectiles/guns/specialist.dm:590]: Seems fine to me.

The meaning of "less balance" is this is effecting a single person, while the previous PR was changing one of the primary marine weapons drastically. So in all, it does in fact, have less balance impact, than the previous PR, which the title is in reference to.

**Comment by Nanu308** [comment]
GA approved, don't touch balance further unless you @ me prior

**Comment by Agameofscones** [comment]
can't approve my own sprites, please get another spriter. Sorry!

**Review by monkeysfist101 — APPROVED** [review]
Sprites LFTM

File: code/modules/projectiles/guns/rifles.dm [hunk 5]
```diff
@@ -1333,13 +1333,73 @@
...
+	accuracy_mult = BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_5
+	accuracy_mult_unwielded = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_4
+	damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_6
+	recoil_unwielded = RECOIL_AMOUNT_TIER_4
+	damage_falloff_mult = 0
+	scatter = SCATTER_AMOUNT_TIER_8
+
+/obj/item/weapon/gun/rifle/m4ra/handle_starting_attachment()
+	..()
+	var/obj/item/attachable/m4ra_barrel/INTEGRATED = new(src)
```
Comment (Zonespace27) [inline code/modules/projectiles/guns/rifles.dm:1388]: why is this in caps? It's not a `CONST` or a `#define`

File: code/modules/projectiles/guns/specialist.dm [hunk 6]
```diff
@@ -520,56 +520,82 @@
...
-	starting_attachment_types = list(/obj/item/attachable/stock/rifle/marksman)
-
+	map_specific_decoration = TRUE
+	wield_delay = WIELD_DELAY_VERY_FAST
+	aim_slowdown = SLOWDOWN_ADS_QUICK
 	flags_item = TWOHANDED|NO_CRYO_STORE
 
+/obj/item/weapon/gun/rifle/m4ra_custom/handle_starting_attachment()
+	..()
+	var/obj/item/attachable/m4ra_barrel_custom/INTEGRATED = new(src)
```
Comment (Zonespace27) [inline code/modules/projectiles/guns/specialist.dm:577]: see above

**Review by Zonespace27 — APPROVED** [review]
nits to be picked; looks good to me otherwise

if nobody else merges this in a few days gimme a ping

**Pull request merged**